src/events/reaction_remove.rs

```diff
@@ -0,0 +1,50 @@
+use super::*;
+
+use serenity::model::channel::ReactionType::Custom;
+
+pub async fn reaction_remove(ctx: client::Context, event: Reaction) -> Result<()> {
+    let user = event.user(&ctx).await?;
+    if user.bot {
+        return Ok(());
+    }
+
+    let msg = event.message(&ctx).await?;
+    if msg
+        .reactions
+        .iter()
+        .find_map(|x| {
+            if x.reaction_type == event.emoji {
+                Some(false)
+            } else {
+                None
+            }
+        })
+        .unwrap_or(true)
```

Comment:
maybe add a comment here? this find_map took me a solid minute to understand the idea of lol

Reply:
Iterator::any

Reply:
uhh, this actually!